File: cp/positive_mex.py
t = int(input())

while(t>0):

    t-=1

    n,s = input().split( )
    n,s = int(n),int(s)

    price = list(map(int,input().split( )))
    flag = list(map(int,input().split( )))

    minf,mind=200,200

    for i in range(n):

        if(flag[i]):
            if(price[i]<minf):
                minf=price[i]

        else:
            if(price[i]<mind):
                mind=price[i]

        i+=1

    if((s+mind+minf)>100):
        print("no")
    else:
        print("yes")

File: cp/positive_mex.cpp
#include<bits/stdc++.h>
#define num long int
using namespace std;

int main()
{
    //ios_base::sync_with_stdio(0);
    //cin.tie(0); cout.tie(0);

    num t;
    cin>>t;
    while(t--)
    {

    num n,s,mind=200,min,minf=200,posd,posf,pos;
    cin>>n>>s;

    num p[n]; bool a[n]; bool flag=1;

    for(num i=0; i<n; i++)
    {
        cin>>p[i]; 
        if(p[i]<minf)
        {
            min=p[i]; pos=i;
        }
    }
    for(num i=0; i<n; i++)
        cin>>a[i];
    if(a[pos]) minf=min;
    else mind = min;
    min = 200;
    if(a[pos])
    for(num i=0; i<n; i++)
    {
        if(a[i]==0 && p[i]<min)
            min=p[i];
    }
    else
    {
        for(num i=0; i<n; i++)
        {
        if(a[i] && p[i]<min)
            min=p[i];
        }
    }
    if(a[pos]) mind=min;
    else minf=min;

    if((s+mind+minf)>100)
        cout<<"no\n";
    else cout<<"yes\n";
    }
    return 0;
}